espflash/src/flasher/mod.rs

```diff
@@ -709,11 +711,18 @@ impl Flasher {
 
         // Load flash stub if enabled
         if use_stub {
-            info!("Using flash stub");
-            flasher.load_stub()?;
+            if !sdm {
```

Comment:
If you bring the `!sdm` if outside, you can avoid two seperate if's here and below